### PR TITLE
LUN-362: Add threshold for finished exercises

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

**Review by f1sh1918 — COMMENTED** [review]
Looks fine for me.
Just recognized one issue probably not connected to that.
1.Enable dev mode -> fail all -> correct finish screen
2. retry solve all -> again failing screen :/
image

File: src/services/helpers.ts [hunk 1]
```diff
@@ -166,3 +165,6 @@ export const calculateScore = (documentsWithResults: DocumentResult[]): number =
       }, 0) / documentsWithResults.length
   )
 }
+
+export const willNextExerciseUnlock = (previousScore: number | undefined, score: number): boolean =>
+  score > SCORE_THRESHOLD_UNLOCK && (previousScore === undefined || previousScore <= SCORE_THRESHOLD_UNLOCK)
```
Comment (steffenkleinle) [inline src/services/helpers.ts:170]: Optional:
```suggestion
  score > SCORE_THRESHOLD_UNLOCK && previousScore ?? 0 <= SCORE_THRESHOLD_UNLOCK)
```

**Review by steffenkleinle — COMMENTED** [review]
Tested on android.

I am getting a warning about a require cycle helpers <-> async storage. Not sure if this was introduced in this PR but definitely related to calculation of the score and should be fixed.

**Comment by steffenkleinle** [comment]
Please add a release note :)

**Comment by SteffiStuffel** [comment]
> Looks fine for me. Just recognized one issue probably not connected to that. 1.Enable dev mode -> fail all -> correct finish screen 2. retry solve all -> again failing screen :/
> 

Actually this was fine, as cheat mode (also in when solving) "needs" three tries, which is to bad to unlock. I adjusted the cheat mode, so it only needs one try when solve all

Removed require cycle, good catch, thanks.

added release note

**Comment by f1sh1918** [comment]
@SteffiStuffel pls resolve merge conflicts and i will rereview

**Review by f1sh1918 — APPROVED** [review]

**Pull request merged**